### PR TITLE
Cursor API and File implementation

### PR DESCRIPTION
**Pull request by swankjesse**
The implementation is easy here; FileInputStream.channel does all of the
work.

### CODE REVIEW

**Comment by swankjesse** [comment]
(stacked PR)

File: okio/src/jvmMain/kotlin/okio/-Platform.kt [hunk 1]
```diff
@@ -52,3 +52,5 @@ actual typealias EOFException = java.io.EOFException
 actual typealias FileNotFoundException = java.io.FileNotFoundException
 
 actual typealias Closeable = java.io.Closeable
+
+actual typealias JvmDefault = kotlin.jvm.JvmDefault
```
Comment (Egorand) [inline okio/src/jvmMain/kotlin/okio/-Platform.kt:56]: It's strange that they don't have it in the common stdlib artifact, unlike some other annotations (https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/). Should we file an issue (unless someone already did it)?

**Review by Egorand — APPROVED** [review]

File: okio/src/jvmMain/kotlin/okio/-Platform.kt (hunk 1 above)
Comment (swankjesse) [reply okio/src/jvmMain/kotlin/okio/-Platform.kt:56]: Yep.

Comment (JakeWharton) [reply okio/src/jvmMain/kotlin/okio/-Platform.kt]: The better approach is to switch to `-Xjvm-default=all` so that you don't need to annotate anything.

Comment (swankjesse) [reply okio/src/jvmMain/kotlin/okio/-Platform.kt]: Oh yes, that's what I want.

**Comment by armay** [comment]
Hi @swankjesse! Happy to see this feature finally arrived! I have a question. Is random access in version 3.0 going to be supported only by sources? I think the case of a sink is similar and it is worth implementing because you can treat a random access file both as a sink and as a source.
Before this PR, I had already been using my own home-made JVM-only implementation of random access source and sink, rather close to yours, in a pet project, and it seems to me like this approach works both ways.
Please, excuse me if random access sink support is already announced somewhere.

**Comment by swankjesse** [comment]
@armay yep! Got a favorite use case for random access writes?

**Comment by armay** [comment]
Great! Thanks for the quick reply! I'll think of a practical example and share it in this thread. My current use case is a bit abstract)

**Pull request merged**